tests/integration/queries/test_migration_progress.py

```diff
@@ -0,0 +1,353 @@
+
+import pytest
+from databricks.sdk import WorkspaceClient
+from databricks.labs.blueprint.wheels import find_project_root
+from databricks.labs.lsql.backends import SqlBackend, Row
+from databricks.labs.lsql.dashboards import DashboardMetadata, Dashboards
+
+from databricks.labs.ucx.progress.install import Historical
+from databricks.labs.ucx.progress.workflow_runs import WorkflowRun
+from databricks.labs.ucx.hive_metastore.grants import Grant
+from databricks.labs.ucx.assessment.jobs import JobInfo
+from databricks.labs.ucx.assessment.clusters import ClusterInfo, PolicyInfo
+from databricks.labs.ucx.assessment.pipelines import PipelineInfo
+from databricks.labs.ucx.hive_metastore.tables import Table
+from databricks.labs.ucx.hive_metastore.table_migration_status import TableMigrationStatus
+from databricks.labs.ucx.hive_metastore.udfs import Udf
+
+from ..conftest import MockInstallationContext
+
+
+@pytest.fixture
+def workflow_runs(ws: WorkspaceClient) -> list[WorkflowRun]:
+    now = dt.datetime.now(tz=dt.timezone.utc).replace(microsecond=0)
+    workspace_id = ws.get_workspace_id()
+    workflow_name = "migration-progress-experimental"
+    workflow_id = 1
+    return [
+        WorkflowRun(
+            started_at=now - dt.timedelta(days=1, minutes=30),
+            finished_at=now - dt.timedelta(days=1),
+            workspace_id=workspace_id,
+            workflow_name=workflow_name,
+            workflow_id=workflow_id,
+            workflow_run_id=1,
+            workflow_run_attempt=1,
+        ),
+        WorkflowRun(
+            started_at=now - dt.timedelta(minutes=30),
+            finished_at=now,
+            workspace_id=workspace_id,
+            workflow_name=workflow_name,
+            workflow_id=workflow_id,
+            workflow_run_id=2,
+            workflow_run_attempt=1,
+        ),
+    ]
+
+
+@pytest.fixture
+def tables():
+    records = []
+    for table in (
+        Table("hive_metastore", schema, table, "MANAGED", "delta")
+        for schema in ("schema1", "schema2")
+        for table in ("table1", "table2", "table3", "table4", "table5")
+    ):
+        record = ("tables", [table.catalog, table.database, table.name], table, [], "Cor")
+        records.append(record)
+    return records
+
+
+@pytest.fixture
+def table_migration_statuses(tables):
+    statuses = []
+    for _, id_, table, _, owner in tables:
+        table_migration_status = TableMigrationStatus(table.catalog, table.database, table.name)
+        failures = ["not migrated"]
+        owner = "Andrew" if table.name == "table1" else "Cor"
+        if table.database == "schema1":  # Simulate one schema being migrated
+            if table.name == "table2":  # An owner with only migrated objects
+                owner = "Eric"
+            table_migration_status.dst_catalog = "catalog1"
+            table_migration_status.dst_schema = table.database
+            table_migration_status.dst_table = table.name
+            failures = []
```

Comment:
@asnare and @nfx : I am incorporating the historical encoder from https://github.com/databrickslabs/ucx/pull/2743, which brings up the question on what the failures are for `TableMigrationStatus`. 

I understood that the intend of the failures is to say:
- if there are failures, then the object is *not* migrated
- The object is migrated otherwise

This means I need to add additional logic in the `Encoder` to do this for `TableMigrationStatus` similar to what I do here: if `dst_` values exists, then no failures

Reply:
failures for migration state should be "table x.y is not yet migrated to a.b.c"

Reply:
covered by https://github.com/databrickslabs/ucx/pull/3083